app/src/main/java/nerd/tuxmobil/fahrplan/congress/details/SessionDetailsViewModel.kt

```diff
@@ -133,60 +69,140 @@ class SessionDetailsViewModel @JvmOverloads constructor(
     val subtitleFont = Font.Roboto.Light
     val titleFont = Font.Roboto.BoldCondensed
 
-    fun onOptionsMenuItemSelected(menuItemId: Int) = when (menuItemId) {
-        R.id.menu_item_feedback -> {
-            val uri = feedbackUrl.toUri()
-            viewActionHandler.openFeedback(uri)
-            true
-        }
-        R.id.menu_item_share_session,
-        R.id.menu_item_share_session_text -> {
-            val formattedSession = session.toPlainText(timeZoneId)
-            viewActionHandler.shareAsPlainText(formattedSession)
-            true
+    val selectedSessionParameter: LiveData<SelectedSessionParameter> = repository.selectedSession
+        .map { it.toSelectedSessionParameter() }
+        .asLiveData(executionContext.database)
+
+    val openFeedBack = SingleLiveEvent<Uri>()
+    val shareSimple = SingleLiveEvent<String>()
+    val shareJson = SingleLiveEvent<String>()
+    val addToCalendar = SingleLiveEvent<Session>()
+    val setAlarm = SingleLiveEvent<Unit>()
+    val navigateToRoom = SingleLiveEvent<Uri>()
+    val closeDetails = SingleLiveEvent<Unit>()
+
+    private fun Session.toSelectedSessionParameter(): SelectedSessionParameter {
+        val useDeviceTimeZone = repository.readUseDeviceTimeZoneEnabled()
+        val formattedZonedDateTime = formattingDelegate.getFormattedDateTimeShort(useDeviceTimeZone, dateUTC, timeZoneOffset)
+        val formattedAbstract = markdownConversion.markdownLinksToHtmlLinks(abstractt)
+        val formattedDescription = markdownConversion.markdownLinksToHtmlLinks(description)
+        // language=regex
+        val linksHtml = getLinks().replace("\\),".toRegex(), ")<br>")
```

Comment:
Could this maybe part of `Session` or better a dedicated `SessionFormatter`? I know it is currently the the only place where it is uses...but would love to extract the responsibility into another place. (But it is not a must for this PR and could also be done in a Follow-up)

Reply:
:heavy_check_mark: Nice, thank you. This gives me the opportunity to increase the unit test scenarios for the particular function. Added in b454420c6317914f52318589c67daf1fd03dffe4.